rand-core/src/lib.rs

```diff
@@ -81,6 +81,14 @@ pub mod le;
 /// in this trait directly, then use the helper functions from the [`impls`]
 /// module to implement the other methods.
 /// 
+/// It is recommended that implementations also implement:
+/// 
+/// - `Debug` but such that no internal details are output
+/// - `Serialize` and `Deserialize` (from Serde), possibly behind a feature gate
+/// - `Clone` if, and only if, the clone will have identical output to the original
```

Comment:
We currently violate this with `ThreadRng`...

Do we want to guarantee the same output? Or do we want to guarantee the same generator?

I think we should just remove `Clone` from `ThreadRng`, but asking anyway. Would be another (small, I think) breaking change.

Reply:
`ThreadRng` is `Clone` because it's a `Rc<..>`. So maybe we should expand the doc to something like:

- if the clone would have identical output (i.e. be a true clone) or if the clone would be another handle to the same generator

Reply:
With that definition `OsRng` could implement `clone` also. I prefer the definition you wrote of identical output.

Reply:
> if the clone would have identical output (i.e. be a true clone) or if the clone would be another handle to the same generator

Does this imply every RNG should document what its clone does? Or should we hide the `Rc` and make `ThreadRng` clone the generator instead of the handle?

Reply:
Technically `ThreadRng` meets the requirement: the clone has identical output to the original, except that the streams are not independent :smile: 

IMO `ThreadRng` supporting `Clone` causes no problems, and might be useful if someone wants to store it in a struct/tuple and then clone that.

Reply:
@vks didn't see your comment until after posting. No, we definitely shouldn't allow cloning the internal `ThreadRng` generator, it's supposed to be a secure way to generate random data, not one which risks returning cloned streams. I don't see much need to document the cloning in general, though I will  for `ThreadRng` because it is an exception.